### PR TITLE
BUG: astype to pyarrow does not copy np array

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

cc @mroeschke 

This should copy. You can't update the pyarrow array, but the parent array can get updated which propagates to the pyarrow array.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
worth adding this to tm.shares_memory?

**Comment by phofl** [comment]
Raises a NotImplementedError (no arrow support)

Will have a look at this in a follow up

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -218,6 +218,9 @@ def _from_sequence(cls, scalars, *, dtype: Dtype | None = None, copy: bool = Fal
         if isinstance(scalars, cls):
             scalars = scalars._data
         elif not isinstance(scalars, (pa.Array, pa.ChunkedArray)):
+            if copy and is_array_like(scalars):
+                # pa array should not get updated when numpy array is updated
+                scalars = scalars.copy()
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:223]: Is it worth specifying `np.array(scalars, copy=True)` manually? There might be an off chance the arraylike doesn't have a `copy` method?

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:223]: This would cast our nullables to object, so not ideal

Comment (mroeschke) [reply pandas/core/arrays/arrow/array.py:223]: Ah gotcha. Would calling `copy.copy(...)` be safer then?

Comment (phofl) [reply pandas/core/arrays/arrow/array.py]: We need deepcopy, but this should do the trick

Comment (jbrockmendel) [reply pandas/core/arrays/arrow/array.py]: > There might be an off chance the arraylike doesn't have a copy method?

might make more sense to raise than guess if this occurs?

Comment (phofl) [reply pandas/core/arrays/arrow/array.py]: No preference here, cc @mroeschke

**Review by mroeschke — APPROVED** [review]

**Pull request merged**